CMR/python/cmr/search/granule.py

```diff
@@ -106,6 +158,39 @@ def search(query, filters = None, limit = None, config: dict = None):
         config=config)
     return found_items
 
+def sample_by_collections(collection_query, filters = None, limits = None, config: dict = None):
+    """
+    Perform a compound search looking for granules based on the results of a
+    collection search. First find a list of collections using a supplied collection
+    query, then take the resulting collection IDs and perform a granule search
+    with them. Granule samples are then returned from many of the collections.
+
+    Parameters:
+        collection_query(dictionary) : a collection query
+        filters(list): a list of filter lambdas which taken in a row and return and row
+        limits: an int, a list of 3 int values or a dictionary, None values will be calculated out
+            int: max limit
+            list: [max, collections, granules]
+            dictionary: {'limit': None, 'collection': None, 'granule': None}
+        config (dictionary): configuration settings
+    """
+    if limits is None:
+        limits = {'limit': None, 'collection': None, 'granule': None}
+    elif isinstance(limits, int):
+        limits = {'limit': limits, 'collection': None, 'granule': None}
+    elif isinstance(limits, list) and len(limits)==3:
+        limits = {'limit': limits[0], 'collection': limits[1], 'granule': limits[2]}
+    else:
+        limits = {'limit': None, 'collection': None, 'granule': None}
+
+    limit, c_limit, g_limit = _collection_sample_limits(limits["limit"],
+        limits["collection"],
+        limits["granule"])
```

Comment:
def _collection_sample_limits(limit:int, climit:int, glimit:int). it says climit is "client" limit, now it's passing "collection" limit.  Please clarify if it's client limit, or collection limit. and name it accordingly.